pkg/controller.v1/tensorflow/tfjob_controller.go

```diff
@@ -499,14 +513,15 @@ func (r *TFJobReconciler) UpdateJobStatus(job interface{}, replicas map[commonv1
 		}
 
 		if failed > 0 {
-			restart := false
-			for _, condition := range jobStatus.Conditions {
-				if condition.Type == commonv1.JobRestarting {
-					restart = true
+			// For the situation that jobStatus has a restarting condition, and appends a new running condition,
+			// the restarting condition will be removed from jobStatus by commonv1.filterOutCondition(),
+			// so we need to append the restarting condition back to jobStatus.
+			if existingRestartingCondition != nil {
+				err := commonutil.UpdateJobConditions(jobStatus, commonv1.JobRestarting, existingRestartingCondition.Reason, existingRestartingCondition.Message)
```

Comment:
Should we reset the value of existingRestartingCondition after this line?

Reply:
Hi @richardsliu, I don't know if we could reset it. I record the existingRestartingCondition outside the loop, which is the condition and result for current `reconcile()`. Inside the loop, when iterate over all pods, if there are multiple workers or masters .etc, are running, and only one worker failed, by the logic inside of `updateJobConditions()`, `restarting` + `running` combination will make `running` condition overwrite `restarting`.

Reply:
@richardsliu is this the right way to deal with it? If one of the pods is restarting, do we mark the whole job as restarting ?

Reply:
If we do something along the lines of, would it achieve the same thing?

```
if condition.Type == commonv1.JobRunning || condition.Type == commonv1.JobRestarting {
  restart = true
```

Reply:
Hi @pavanky, I would like to hear more advices: ) For now, I just follow the current convention. See https://github.com/kubeflow/training-operator/blob/master/pkg/controller.v1/tensorflow/tfjob_controller.go#L762 . Current strategy is that if one of the pods failed, then update the whole job status to failed : /

Reply:
Wouldn't that technically mean we are updating the job status to restart and increment two times for the same failure ?

Reply:
IMO, we could see it from 2 points of views.
1. If jobStatus is `running`, and one of roles got failed pods, then the whole job will be updated to `failed`, then in next reconcile, the pods will be deleted, so jobstatus should have a `restarting` (don't know if I understand the code base correctly)
2. Technically we just set jobstatus's field twice without sending request to apiServer, so in fact, it doesn't got really updated. But yeah, current one is not elegant.